### PR TITLE
[medium] more granular permissions for investigators

### PR DESCRIPTION
**Pull request by ameihm0912**
This removes the previous implementation where an investigator could
either be set to an admin or not, and expands this to per-endpoint
permissions that can be set for individual investigators.

### CODE REVIEW

**Comment by ameihm0912** [comment]
@jvehent r?

**Comment by ameihm0912** [comment]
Updated with more fine-grained privileges per-endpoint

The UI component needs some expansion to support the granularity and currently only operates on categories/selection macros

File: mig-api/api.go [hunk 1]
```diff
@@ -250,7 +238,10 @@ func authenticate(pass handler, adminRequired bool) handler {
 		if !ctx.Authentication.Enabled {
 			inv.Name = "authdisabled"
 			inv.ID = 0
-			inv.IsAdmin = true
+			inv.Permissions.DefaultSet()
+			inv.Permissions.ManifestSet()
+			inv.Permissions.LoaderSet()
+			inv.Permissions.AdminSet()
```
Comment (jvehent) [inline mig-api/api.go]: Should it just be `0xFFFFFFFFFFFFFFFF` to catch future permissions as well?

Comment (ameihm0912) [inline mig-api/api.go]: Yeah good point, we could do something like inv.Permissions.FromMask(0xffffffffffffffff)

**Pull request merged**